Library/Homebrew/cask/audit.rb

```diff
@@ -540,7 +541,34 @@ def audit_livecheck_version
       false
     end
 
-    def audit_livecheck_min_os
+    sig { void }
+    def audit_min_os
+      return unless online?
+      return unless strict?
+
+      odebug "Auditing minimum OS version"
+
+      sparkle_min_os = livecheck_min_os
+      plist_min_os = cask_plist_min_os
+      odebug "Minimum OS version: Plist #{plist_min_os} | Sparkle #{sparkle_min_os.inspect}"
+      min_os_string = [sparkle_min_os, plist_min_os].compact.max
```

Comment:
I don't think `.max` will work like how you want/expect on two strings. You probably need to do `Version.new` on them both and then compare them instead. String version comparisons can look like they work sometimes but are not worth relying on.

Reply:
@MikeMcQuaid, I did not find great support for the partial extraction. I tried to deduplicate the extraction operation that was duplicated in the signing audit and the minimum OS audit. 

Both livecheck_min_os and plist_min_os are a `MacOSVersion`.